### PR TITLE
feat: add WnTimestamp component

### LINKED ISSUE #149: Component "Time Stamp"
Designs and description:
https://www.figma.com/design/J9pCZpUhcm0MRs7dFX7LTN/01.-Application-Components?node-id=835-1965&t=PG1G9rnnb536UAVJ-1

### PR DESCRIPTION
**Pull request by erskingardner**
## Summary

Closes #149

- Add `WnTimestamp` widget that displays intelligent relative/absolute timestamps for chat list
- Format adapts based on message age: Now → minutes → hours → time → yesterday → weekday → date → date with year
- Localized for 8 languages (EN, DE, ES, FR, IT, PT, RU, TR)
- Includes widgetbook showcase with interactive date/time picker playground

## Timestamp Formats

| Age | Format | Example |
|-----|--------|---------|
| < 60 seconds | Now | `Now` |
| 1-59 minutes | Minutes | `32m` |
| 1-12 hours | Hours | `4h` |
| Same day, 12+ hours | Time | `14:30` |
| Previous day, 12+ hours | Yesterday | `Yesterday` |
| 2-6 days ago | Weekday | `Monday` |
| > 6 days, < 1 year | Date | `Jun 10` |
| > 1 year | Date & Year | `Jan 9, 2009` |

## Testing

- 39 comprehensive tests covering all formatting rules and edge cases
- All precommit checks pass

## Summary by CodeRabbit

* **New Features**
  * Added a timestamp widget that displays human-friendly relative times, weekdays, and date formats.
  * Added an interactive timestamp showcase/playground and a DateTime picker knob in the component explorer.
  * Added a utility for formatting DateTime for display.

* **Localization**
  * Extended time- and calendar-related translations across many locales (now, minutes, hours, yesterday, weekdays, short month names).
  * Enabled localization support in the component explorer.

* **Tests**
  * Added comprehensive tests covering formats, boundaries, and styling.

### CODE REVIEW

File: widgetbook/lib/components/timestamp.dart [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+          'Playground',
+          style: TextStyle(
+            fontSize: 20,
+            fontWeight: FontWeight.w600,
+            color: context.colors.backgroundContentPrimary,
+          ),
+        ),
+        const SizedBox(height: 8),
+        Text(
+          'Use the knobs panel to customize the timestamp.',
```
[the remark this answers is not in the document]
Comment (josefinalliende) [reply widgetbook/lib/components/timestamp.dart]: https://github.com/marmot-protocol/sloth/pull/177
